pkg/kubectl/cmd/attach.go

```diff
@@ -167,7 +167,7 @@ func (p *AttachOptions) Run() error {
 		if err != nil {
 			return err
 		}
-		if pod.Status.Phase != api.PodRunning {
+		if pod.Status.Phase != api.PodRunning && (pod.Status.Phase == api.PodPending && len(pod.Spec.InitContainers) < 1) {
```

Comment:
So should I change it to include conditions from https://github.com/kubernetes/kubernetes/pull/27359/files#diff-de6f7fc4d605130323893d0b7572045eR177 ? @sttts @smarterclayton ptal